krun/scheduler.py

```diff
@@ -370,25 +370,30 @@ def run(self, mailer, dry_run=False):
         if not dry_run:
             try:
                 measurements = util.check_and_parse_execution_results(
-                    stdout, stderr, rc)
+                    stdout, stderr, rc, self.sched.config)
                 flag = "C"
+            except util.RerunExecution as e:
+                util.log_and_mail(mailer, warn,
+                                  "Benchmark needs to be re-run: %s" % self.key,
```

Comment:
At this level the exception message (which makes the message body) says why. The exception type itself is intentionally generic in case other re-run scenarios arise later.